scripts/gitlab_runner_pre_build/pre_build.py

```diff
@@ -0,0 +1,70 @@
+"""
+This script is responsible for taking the GITLAB_OIDC_TOKEN from the environment and
+exchanging it for temporary AWS credentials. These credentials are then printed to stdout to be
+sourced by the gitlab runner in its pre_build configuration option.
+
+In the case of a PR build, the temporary credentials are scoped down to only allow access to the
+S3 bucket prefix for the relevant PR.
+"""
+import boto3, os, sys, jwt, json
+
+TEMPORARY_CREDENTIALS_DURATION = 3600 * 6  # 6 hours
+
+if __name__ == "__main__":
+    if not os.environ.get("GITLAB_OIDC_TOKEN"):
+        print("GITLAB_OIDC_TOKEN not in the environment", file=sys.stderr)
+        sys.exit(0)  # this isn't an error yet.
```

Comment:
Meaning: This isn't an error yet because we haven't migrated all the runners to use this approach, but it will be an error once we get to that point?

Reply:
Yes.